### ISSUE TITLE
/sys/bus/iio/devices empty?

### ISSUE THREAD

**Issue by alexhulbert** [jimmy00784/InspironAutoRotateService#3]
Hi! I'd love to use this, but my iio/devices directory is empty. Did you have to do anything beforehand to get the gyroscope to pop up?

**Comment by lalanikarim** [comment]
@Taconut What is the make/model of your laptop and what distro are you using? The scripts were created and tested on Dell Inspiron convertible laptops. 
I am on ArchLinux and I didn't have to install any special drivers.

**Comment by alexhulbert** [comment]
I'm using the Inspiron 7779 on Ubuntu 16.04. Perhaps the difference in distro is the root of the issue? Although a lot could change even with a small change in the model of the laptop.

**Comment by lalanikarim** [comment]
@Taconut 
The gyroscope data is populated by the "hid_sensor_accel_3d" module. As a first step, try lsmod and search to see if you have any modules starting with "hid_sensor_".
I am not familiar as to where Ubuntu might make available the virtual file for the gyroscope and other probes. You might look into /dev directory to see if and special files start with name "iio:device" as they appear under Arch. If the accel_3d module is loaded and you see the iio:device* entries under /dev, then it's a matter of locating the where the device folder might be created under Ubuntu and updating the inspironAutoRotateService accordingly.
Let me know if this helps.

**Comment by alexhulbert** [comment]
@jimmy00784: Thanks for your help! My iio/devices is no longer empty after updating my kernel to 4.12! I did a little bit of debugging and it looks like there were two additional problems:
 - The Systemd service that starts /usr/local/bin/inspironRotateService wasn't working
 - inspironRotateScreen doesn't work

I've fixed the first issue and I'm in the process of fixing the second, but I'll go ahead and close this issue since all my issues seem to be resolved.

Thanks again for your help!
